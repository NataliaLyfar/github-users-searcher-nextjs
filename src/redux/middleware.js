import { githubUsersApi } from "./githubUsers/githubUsersApi";
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from "redux-persist";

export const middleware = (getDefaultMiddleware) => [
  ...getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }),
  githubUsersApi.middleware,
];
