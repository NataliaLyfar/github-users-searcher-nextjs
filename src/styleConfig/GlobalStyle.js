import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
html {
  box-sizing: border-box;
  width: 100vw;
  overflow-x: hidden;
}
*,
*::before,
*::after {
  box-sizing: inherit;
};
body {
  position: relative;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif; 
    color: #000;
};
ul {
  list-style: none;
  padding: 0;
};
li.selected {
background-color: #eead71;
}
div.swiper-button-next::after{
  color: #eead71 !important;
}
div.swiper-button-prev::after{
  color: #eead71 !important;
}
p, h1, h2, h3, h4, button {
  margin: 0;
  padding: 0;
};
img {
  display: block;
  max-width: 100%;
  height: auto;
};
a {
  text-decoration: none;
};
img.ant-image-preview-img {
  margin: 0 auto;
}
`;
