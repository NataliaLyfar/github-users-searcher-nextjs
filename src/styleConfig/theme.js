export const theme = {
  colors: {
    black: '#000',
    white: '#fff',
    primary: 'rgba(0, 0, 0, 0.85)',
    shade: '#d9d9d9',
    accent: '#1890ff',
    hover: '#40a9ff',
  },
  space: [5, 8, 12, 16, 20, 25, 32, 50, 64, 88, 98, 120, 180],
  fonts: {
    body: 'Roboto Slab, serif',
  },
  fontSizes: {
    xs: '12px',
    s: '14px',
    m: '16px',
    l: '18px',
    xl: '25px',
    xxl: '32px',
  },
  fontWeights: {
    normal: 400,
    semibold: 500,
    bold: 700,
  },
  lineHeights: {
    body: 1.5715,
  },
  borders: {
    none: 'none',
    normal: '1px solid',
  },
  radii: {
    small: '2px',
    normal: '5px',
    large: '10px',
    round: '50%',
  },
};
