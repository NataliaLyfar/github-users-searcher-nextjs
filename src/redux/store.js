import { configureStore } from "@reduxjs/toolkit";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import { middleware } from "./middleware";
import { searchSlice, usersSlice } from "./githubUsers/githubUsersSlice";
import { githubUsersApi } from "./githubUsers/githubUsersApi";

const githubUsersPersistConfig = {
  key: "users",
  storage,
  whiteList: ["users"],
};

export const store = configureStore({
  reducer: {
    [githubUsersApi.reducerPath]: githubUsersApi.reducer,
    [usersSlice.name]: persistReducer(
      githubUsersPersistConfig,
      usersSlice.reducer
    ),
    [searchSlice.name]: searchSlice.reducer,
  },
  middleware,
});

export const persistor = persistStore(store);
