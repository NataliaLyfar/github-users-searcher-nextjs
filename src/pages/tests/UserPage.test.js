import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";

import { store } from "../../redux/store";
import { theme } from "../../styleConfig/theme";
import User, { getServerSideProps } from "../users/[userLogin]";

const testUser = {
  avatar_url: "https://avatars.githubusercontent.com/u/2738851?v=4",
  bio: null,
  blog: "http://www.fatihgursoy.com",
  company: null,
  created_at: "2012-11-06T22:55:09Z",
  email: null,
  events_url: "https://api.github.com/users/fg/events{/privacy}",
  followers: 60,
  followers_url: "https://api.github.com/users/fg/followers",
  following: 85,
  following_url: "https://api.github.com/users/fg/following{/other_user}",
  gists_url: "https://api.github.com/users/fg/gists{/gist_id}",
  gravatar_id: "",
  hireable: true,
  html_url: "https://github.com/fg",
  id: 2738851,
  location: "Istanbul, Turkey",
  login: "fg",
  name: "Fatih GÃœRSOY",
  node_id: "MDQ6VXNlcjI3Mzg4NTE=",
  organizations_url: "https://api.github.com/users/fg/orgs",
  public_gists: 0,
  public_repos: 86,
  received_events_url: "https://api.github.com/users/fg/received_events",
  repos_url: "https://api.github.com/users/fg/repos",
  site_admin: false,
  starred_url: "https://api.github.com/users/fg/starred{/owner}{/repo}",
  subscriptions_url: "https://api.github.com/users/fg/subscriptions",
  twitter_username: null,
  type: "User",
  updated_at: "2022-10-14T12:23:36Z",
  url: "https://api.github.com/users/fg",
};

describe("getServerSideProps()", () => {
  window.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve(testUser),
    })
  );
  it("should redirect when error", async () => {
    const context = {
      params: { userLogin: "fg[[[" },
    };
    const response = await getServerSideProps(context);
    expect(response).toEqual(
      expect.objectContaining({
        redirect: { destination: "/" },
      })
    );
  });
  it("should call api", async () => {
    const context = {
      params: { userLogin: "fg" },
    };
    const response = await getServerSideProps(context);
    expect(response).toEqual(
      expect.objectContaining({
        props: { user: testUser },
      })
    );
  });
});
jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/users",
      pathname: "/users",
      query: { userLogin: "fg" },
      asPath: "/users/fg",
    };
  },
}));
describe("UserPage", () => {
  it("UserPage Should render correctly on route: /users/fg", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <User user={testUser} />
        </Provider>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
