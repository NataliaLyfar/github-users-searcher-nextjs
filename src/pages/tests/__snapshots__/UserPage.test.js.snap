// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPage UserPage Should render correctly on route: /users/fg 1`] = `
<div>
  <main>
    <div
      class="sc-dkrFOg sc-eDvSVe bhpLZD iCIKSH"
    >
      <button
        class="ant-btn ant-btn-default ant-btn-icon-only sc-hLBbgP itRRes"
        type="button"
      >
        <span
          aria-label="arrow-left"
          class="anticon anticon-arrow-left"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="arrow-left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
      </button>
      <div
        class="ant-card ant-card-bordered"
        style="width: 320px;"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="user avatar"
            src="https://avatars.githubusercontent.com/u/2738851?v=4"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <button
            class="sc-bcXHqe bqUnOY"
            role="button"
          >
            <span
              aria-label="heart"
              class="anticon anticon-heart"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="heart"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
                />
              </svg>
            </span>
          </button>
          <h3
            class="ant-typography"
          >
            Fatih GÃœRSOY
          </h3>
          <div
            class="ant-typography"
          >
            <span
              class="ant-typography"
            >
              <strong>
                Login:
              </strong>
            </span>
             
            fg
          </div>
          <div
            class="ant-typography"
          >
            <span
              class="ant-typography"
            >
              <strong>
                Followers:
              </strong>
            </span>
             
            60
          </div>
          <div
            class="ant-typography"
          >
            <span
              class="ant-typography"
            >
              <strong>
                Following:
              </strong>
            </span>
             
            85
          </div>
          <div
            class="ant-typography"
          >
            <span
              class="ant-typography"
            >
              <strong>
                Public repos:
              </strong>
            </span>
             
            86
          </div>
          <a
            class="ant-typography"
            href="https://github.com/fg"
            rel="noreferrer"
            target="_blank"
          >
            https://github.com/fg
          </a>
        </div>
      </div>
    </div>
  </main>
</div>
`;
